shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/SchemaBuilder.java

```diff
@@ -130,15 +141,30 @@ private static void appendRemainTables(final SchemaBuilderMaterials materials, f
         return Optional.empty();
     }
     
-    private static void appendDialectRemainTables(final DialectTableMetaDataLoader dialectLoader, final SchemaBuilderMaterials materials, final Map<String, TableMetaData> tables) throws SQLException {
+    private static void appendDialectRemainTables(final DialectTableMetaDataLoader dialectLoader,
+            final SchemaBuilderMaterials materials, final Map<String, TableMetaData> tables, final Collection<String> ruleLogicTables) throws SQLException {
+
+        Map<String, String> actualTable2LogicTableMap = getActualTable2LogicTableMap(materials, ruleLogicTables);
+
         Collection<Future<Map<String, TableMetaData>>> futures = new LinkedList<>();
-        Collection<String> existedTables = getExistedTables(materials.getRules(), tables);
         for (DataSource each : materials.getDataSourceMap().values()) {
-            futures.add(EXECUTOR_SERVICE.submit(() -> dialectLoader.load(each, existedTables)));
+            futures.add(EXECUTOR_SERVICE.submit(() -> dialectLoader.load(each, Collections.emptyList())));
         }
         for (Future<Map<String, TableMetaData>> each : futures) {
             try {
-                tables.putAll(each.get());
+                Map<String, TableMetaData> tableMetaMap = each.get();
+                for (Map.Entry<String, TableMetaData> one : tableMetaMap.entrySet()) {
+                    String actualTableName = one.getKey();
+                    String logicTableName = actualTable2LogicTableMap.get(actualTableName);
```

Comment:
Maybe it's better to extract this code into a method.

Reply:
OK